sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -261,35 +261,33 @@ class kernel_bundle_impl {
     const auto DevImgIt =
```

Comment:
NIT, Consider moving the loop before `unique` and `sort` to make it more easy to understand that `erase` happens as part of `sort + unique + erase` sequence.

Reply:
I'll submit it as a separate patch